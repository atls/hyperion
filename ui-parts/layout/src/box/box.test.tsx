/**
 * @jest-environment jsdom
 */

import { render } from '@testing-library/react'

import React      from 'react'

import { Box }    from './index'

describe('Box component', () => {
  describe('snapshots', () => {
    it('should match latest render snapshot', () => {
      const { asFragment } = render(<Box />)
      expect(asFragment()).toMatchSnapshot()
    })
  })

  it('applies default styles', () => {
    const { container } = render(<Box />)
    const boxElement = container.firstChild as HTMLElement

    const computedStyles = window.getComputedStyle(boxElement)
    expect(computedStyles.getPropertyValue('display')).toBe('flex')
    expect(computedStyles.getPropertyValue('box-sizing')).toBe('border-box')
  })

  it('applies fill styles when fill prop is true', () => {
    const { container } = render(<Box fill />)
    const boxElement = container.firstChild as HTMLElement

    const computedStyles = window.getComputedStyle(boxElement)
    expect(computedStyles.getPropertyValue('width')).toBe('100%')
    expect(computedStyles.getPropertyValue('height')).toBe('100%')
  })

  it('applies fullViewportHeight styles when fullViewportHeight prop is true', () => {
    const { container } = render(<Box fullViewportHeight />)
    const boxElement = container.firstChild as HTMLElement

    const computedStyles = window.getComputedStyle(boxElement)
    expect(computedStyles.getPropertyValue('height')).toBe('100vh')
  })

  it('applies fullViewportWidth styles when fullViewportWidth prop is true', () => {
    const { container } = render(<Box fullViewportWidth />)
    const boxElement = container.firstChild as HTMLElement

    const computedStyles = window.getComputedStyle(boxElement)
    expect(computedStyles.getPropertyValue('width')).toBe('100vw')
  })
})
