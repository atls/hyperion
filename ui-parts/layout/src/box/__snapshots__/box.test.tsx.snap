// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Box component snapshots should match latest render snapshot 1`] = `
<DocumentFragment>
  .emotion-0 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

<div
    class="emotion-0"
    display="flex"
  />
</DocumentFragment>
`;
